### PR TITLE
Ask Save when Closing Project

### LINKED ISSUE #145: [Feature Request] Add save reminder when closing the app.
### Validations

- [X] I have checked for similar feature requests and could not find any.
- [X] I have made sure this is not an already-existing feature.

### Description

When closing Creator, there is no reminder/question to save the current file before closing. Adding this feature will help forgetful people like me very much!

### Suggested solution

Add a confirmation before Creator is closed, asking whether the user want to save the file first or not. The detection can be taken from History panel, for example (if this panel is not empty, then we assume that the file has been modified and then asks for saving before closing).

### Alternative solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by r888800009**
* resolves https://github.com/Inochi2D/inochi-creator/issues/145
* Allow users to configure exit behavior
* A window pops up when the project is closed, exited, or new.
  * NewProjectAskHandler, CloseAskHandler, ExitAskHandler
* Import pop-up windows is currently not supported. I think `kra.d` and `psd.d` shoud be refactored first.
  * https://github.com/nijigenerate/nijigenerate/pull/30
  * https://github.com/nijigenerate/nijigenerate/pull/29
* Pop-up window when "open" is not currently supported

### CODE REVIEW

File: source/creator/io/save.d [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    void onProjectClose() {
+        // override this function for different close behaviour
+        incCloseProject();
+    }
+
+    void show() {
+        incDialog(
+            INC_EXIT_ASK_DIALOG_NAME,
+            __("Save changes before close?"),
+            _("Would you like to save your changes before closing?\n\nYou can change the default behaviour in the settings."),
```
Comment (LunaTheFoxgirl) [inline source/creator/io/save.d:169]: Maybe these should be reworded slightly;
Title should be "Save before quitting?"
Body should be "Would you like to save your changes before quitting?\n\nYou can change the default behaviour in the settings."

File: source/creator/windows/settings.d [hunk 2]
```diff
@@ -195,25 +195,41 @@ protected:
...
+                            string selected = option.get(incGetSaveProjectOnClose(), "Ask");
+                            if (igBeginCombo(__("Save project"), selected.toStringz)) {
+                                foreach(key, displayName ; option) {
+                                    if (igSelectable(displayName.toStringz, selected == key))
+                                        incSetSaveProjectOnClose(key);
+                                }
 
+                                igEndCombo();
+                            }
+                            incTooltip(_("When closing a project, should it be saved automatically?"));
```
Comment (LunaTheFoxgirl) [inline source/creator/windows/settings.d]: Wording should be: "Should changes be saved automatically when closing a project?"

**Review by LunaTheFoxgirl — CHANGES_REQUESTED** [review]
Wording should be fixed, but otherwise looks good.

File: source/creator/io/save.d (hunk 1 above)
Comment (r888800009) [reply source/creator/io/save.d:169]: I think "closing" is more appropriate for various scenarios.

This window is used repeatedly when exiting, opening a new project, or closing an existing project.
Maybe we should change title to "Save before closing?"

Comment (r888800009) [reply source/creator/io/save.d:169]: Or we can implement different windows for different cases

Comment (LunaTheFoxgirl) [reply source/creator/io/save.d:169]: Different windows for different cases seems like a lot of duplicated work.
This will do for now.

**Review by LunaTheFoxgirl — APPROVED** [review]

**Pull request merged**